jaxdecomp/_src/transpose.py

```diff
@@ -44,16 +44,21 @@ def abstract(x, kind, pdims, global_shape):
     assert kind in ['x_y', 'y_z', 'z_y', 'y_x']
     match kind:
     # From X to Y the axis are rolled by 1 and pdims are swapped wrt to the input pdims
-      case 'x_y' | 'y_z':
+      case 'x_y':
         transpose_shape = (2, 0, 1)
-      case 'y_x' | 'z_y':
+        transpose_pdims = pdims[::-1]
+      case 'y_z':
+        transpose_shape = (2, 0, 1)
+        transpose_pdims = pdims
+      case 'z_y':
         transpose_shape = (1, 2, 0)
+        transpose_pdims = pdims[::-1]
+      case 'y_x':
+        transpose_shape = (1, 2, 0)
+        transpose_pdims = pdims
 
```

Comment:
A forward transpose cycle shifts the axis right for us (2 0 1)
A backward transpose cycle shifts the axis right for us (1 2 0)

Every transpose the pdims are switched (implicitly by cudecomp)
I need the original pdims for the grid descriptor
So I reverse the pdims every other transpose